iris-mpc-cpu/src/hnsw/sorting/batcher.rs

```diff
@@ -0,0 +1,239 @@
+use super::sorting_network::{SortingNetwork, SortingNetworkLayer};
+
+/// Generate sort tuples for specified stage of batcher merge network
+fn batcher_merge_step(deg: usize, stage: usize) -> SortingNetworkLayer {
+    let (offset, scale) = (1 << (deg - stage), 1 << (stage - 1));
```

Comment:
perhaps, it would be easier to read this function if the special case `stage = 1` is handled right at the start, i.e.,
```rust
let offset = 1 << (deg - stage);
if stage == 1 {
    return (0..offset).map(|start| (start, start + offset)).collect();
}
...
`